pkg/kubelet/kubelet.go

```diff
@@ -439,12 +439,22 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	}
 
 	serviceIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	var serviceListers []corelisters.ServiceLister
 	if hasValidTPClients(kubeDeps.KubeTPClients) {
-		serviceLW := cache.NewListWatchFromClient(kubeDeps.KubeTPClients[0].CoreV1(), "services", metav1.NamespaceAll, fields.Everything())
-		r := cache.NewReflector(serviceLW, &v1.Service{}, serviceIndexer, 0)
-		go r.Run(wait.NeverStop)
+		serviceListers = make([]corelisters.ServiceLister, len(kubeDeps.KubeTPClients))
+
+		for i := range serviceListers {
+			serviceLW := cache.NewListWatchFromClient(kubeDeps.KubeTPClients[i].CoreV1(), "services", metav1.NamespaceAll, fields.Everything())
+			r := cache.NewReflector(serviceLW, &v1.Service{}, serviceIndexer, 0)
+			go r.Run(wait.NeverStop)
+
+			serviceListers[i] = corelisters.NewServiceLister(serviceIndexer)
+		}
+	} else {
+		klog.Errorf("No Valid TP Clients: %v", err)
```

Comment:
if the hasValidTPclient() returns false, then the kubeclient is nil or 0 length. this else block should return error.